### PR TITLE
Perform recommendations and fixes from internal audit on AddressBook app 

### PR DESCRIPTION
**Pull request by ottodevs**
Skipped:
- Registry optimizations
- Better IPFS params type checks on contract
- Attack vectors on UI side related to IPFS checks or content
- Tests: further address registry edge scenarios

### CODE REVIEW

File: apps/address-book/contracts/AddressBook.sol [hunk 1]
```diff
@@ -1,86 +1,104 @@
...
      * @param _addr The ID of the entry to remove
+     * @param _cid The IPFS hash of the entry to remove from the registry; used only for radpec here
      */
-    function removeEntry(
-        address _addr
-    ) public auth(REMOVE_ENTRY_ROLE)
-    {
-        require(bytes(entries[_addr]).length != 0, "entry does not exist");
-
+    function removeEntry(address _addr, string _cid) public auth(REMOVE_ENTRY_ROLE) entryExists(_addr) {
```
Comment (Quazia) [inline apps/address-book/contracts/AddressBook.sol]: If we're going to reference the CID in the radspec, and we expect it to be meaningful we need to require the CID matches the current CID for this entry.

**Comment by ottodevs** [comment]
Thanks for making the change @topocount 💯

**Comment by ottodevs** [comment]
@Quazia do you feel there is something missing to address for this one?

**Comment by ottodevs** [comment]
The base branch was changed to mainnet-audit since is the main branch to merge the internal changes, instead of the old target mainnet-rewards-audit-1 that is already merged!

Some conflicts arise with this change, but should be easily fixable.

**Comment by ottodevs** [comment]
We can close this, following @topocount guideline from #1155

**Pull request closed without merging**